SpeedyTransforms/src/fourier.jl

```diff
@@ -61,16 +56,9 @@ function _apply_batched_fft!(
     nlayers = size(field, 2)        # number of vertical layers
     nfreq = nlon÷2 + 1              
 
-    # Construct views of the data to perform the FFT on
-    # views and copies necessary for stride-1 outputs required by FFTW
-    out = reshape(view(S.scratch_memory_grid, 1:nlon*nlayers), (nlon, nlayers))
-
-    # PERFORM FFT, inverse complex to real, hence brfft
-    # FFTW is in-place writing into `out` via `mul`
-    # FFTW requires stride-1 output, hence view on scratch memory
     if not_equator  # skip FFT, redundant when north already did that latitude
-        LinearAlgebra.mul!(out, brfft_plan, view(g_in, 1:nfreq, 1:nlayers, j))
-        field[ilons, :] = out
+        # this version currently allocates a small array, FFTW supports strided outputs, but FFTW.jl doesn't 
```

Comment:
for every latitude this is a length nlayer vector correct?

Reply:
I think it's ~ `nlat * nlayers * nfreq` all together for the complete transform, no? with `nfreq` and `nlat` varying per ring.